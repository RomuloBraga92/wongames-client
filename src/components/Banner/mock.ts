export default {
  img: 'https://source.unsplash.com/user/willianjusten/1042x580',
  title: 'Defy death',
  subtitle: '<p>Play the new <strong>CrashLands</strong> season',
  buttonLabel: 'Buy now',
  buttonLink: '/games/defy-death',
}
