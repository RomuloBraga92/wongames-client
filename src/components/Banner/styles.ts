import styled from 'styled-components'

type ImageProps = {
  src: string
}

export const Container = styled.main``

export const Image = styled.div<ImageProps>``

export const BannerInfo = styled.div``

export const Title = styled.h2``

export const Subtitle = styled.h3``
