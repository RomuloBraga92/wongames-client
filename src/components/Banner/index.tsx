import Button from 'components/Button'
import { RibbonColors, RibbonSizes } from 'components/Ribbon'
import Ribbon from 'components/Ribbon'
import * as S from './styles'

export type BannerProps = {
  img: string
  title: string
  subtitle: string
  buttonLabel: string
  buttonLink: string
  ribbon?: React.ReactNode
  ribbonColor?: RibbonColors
  ribbonSize?: RibbonSizes
}

const Banner = ({
  img,
  title,
  subtitle,
  buttonLabel,
  buttonLink,
  ribbon,
  ribbonColor = 'primary',
  ribbonSize = 'normal',
}: BannerProps) => (
  <S.Container>
    <S.Image src={img} role="img" aria-label={title} />
    {!!ribbon && (
      <Ribbon color={ribbonColor} size={ribbonSize}>
        {ribbon}
      </Ribbon>
    )}

    <S.BannerInfo>
      <S.Title>{title}</S.Title>
      <S.Subtitle dangerouslySetInnerHTML={{ __html: subtitle }} />
      <Button as="a" href={buttonLink} size="large">
        {buttonLabel}
      </Button>
    </S.BannerInfo>
  </S.Container>
)

export default Banner
