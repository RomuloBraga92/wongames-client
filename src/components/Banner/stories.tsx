import { Story, Meta } from '@storybook/react/types-6-0'
import Banner, { BannerProps } from '.'
import item from './mock'

export default {
  title: 'Banner',
  component: Banner,
  argTypes: {
    ribbon: {
      type: 'string',
    },
  },
  args: { ...item },
  parameters: {
    layout: 'fullscreen',
  },
} as Meta

export const Default: Story<BannerProps> = args => (
  <div style={{ maxWidth: '104rem', margin: '0 auto' }}>
    <Banner {...args} />
  </div>
)

export const WithRibbon: Story<BannerProps> = args => (
  <div style={{ maxWidth: '104rem', margin: '0 auto' }}>
    <Banner {...args} />
  </div>
)

WithRibbon.args = {
  ribbon: '20% OFF',
  ribbonSize: 'normal',
  ribbonColor: 'primary',
}
