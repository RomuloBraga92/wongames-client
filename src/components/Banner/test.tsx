import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Banner from './'

describe('<Banner />', () => {
  beforeEach(() => {
    renderWithTheme(
      <>
        <Banner
          img="https://source.unsplash.com/user/willianjusten/1042x580"
          title="Defy death"
          subtitle="<p>Play the new <strong>CrashLands</strong> season"
          buttonLabel="Buy now"
          buttonLink="/games/defy-death"
        />
      </>,
    )
  })

  it('should render a title', () => {
    expect(
      screen.getByRole('heading', { name: /Defy death/i }),
    ).toBeInTheDocument()
  })
  it('shound render a subtitle', () => {
    expect(
      screen.getByRole('heading', { name: /Play the new crashlands season/i }),
    ).toBeInTheDocument()
  })
  it('should render an image', () => {
    expect(screen.getByRole('img', { name: /Defy death/i })).toBeInTheDocument()
  })
})
