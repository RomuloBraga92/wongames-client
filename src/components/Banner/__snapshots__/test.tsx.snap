// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    Banner
  </h1>
</main>
`;
